frontend/taipy-gui/src/components/Taipy/TimeSelector.spec.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2021-2024 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+import React from "react";
+import { render } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import "@testing-library/jest-dom";
+
+import { LocalizationProvider } from "@mui/x-date-pickers";
+import { AdapterDateFns } from "@mui/x-date-pickers/AdapterDateFnsV3";
+
+import TimeSelector from "./TimeSelector";
+import { TaipyContext } from "../../context/taipyContext";
+import { TaipyState, INITIAL_STATE } from "../../context/taipyReducers";
+import { getClientServerTimeZoneOffset } from "../../utils";
+
+
+jest.mock("../../utils", () => {
+    const originalModule = jest.requireActual("../../utils");
+
+    //Mock getClientServerTimeZoneOffset
+    return {
+        __esModule: true,
+        ...originalModule,
+        getClientServerTimeZoneOffset: () => 0,
+    };
+});
+
+beforeEach(() => {
+    // add window.matchMedia
+    // this is necessary for the date picker to be rendered in desktop mode.
+    // if this is not provided, the mobile mode is rendered, which might lead to unexpected behavior
+    Object.defineProperty(window, "matchMedia", {
+        writable: true,
+        value: (query: string): MediaQueryList => ({
+            media: query,
+            // this is the media query that @material-ui/pickers uses to determine if a device is a desktop device
+            matches: query === "(pointer: fine)",
+            onchange: () => {},
+            addEventListener: () => {},
+            removeEventListener: () => {},
+            addListener: () => {},
+            removeListener: () => {},
+            dispatchEvent: () => false,
+        }),
+    });
+});
+
+afterEach(() => {
+    // @ts-ignore
+    delete window.matchMedia;
+});
+
+const curDate = new Date();
+curDate.setHours(1, 1, 1, 1);
+const curDateStr = curDate.toISOString();
+
+const cleanText = (val: string) => val.replace(/\u200e|\u2066|\u2067|\u2068|\u2069/g, "");
+
+describe("TimeSelector component", () => {
+    it("renders", async () => {
+        const { getByTestId } = render(
+            <LocalizationProvider dateAdapter={AdapterDateFns}>
+                <TimeSelector time={curDateStr} />
+            </LocalizationProvider>
+        );
+        const elt = getByTestId("ClockIcon");
+        expect(elt.parentElement?.tagName).toBe("BUTTON");
+    });
+    it("displays picker as a clock", async () => {
```

Comment:
Is there something more specific for asClock?

Reply:
Do u mean i have to make it into separate set of unit tests like `with_time`?